guardian/editions
# Delete expired editions

## Summary
This PR adds a job, which runs every time the app starts, to clean up editions that no longer exist in the `editionsListCache` (async storage). The main motivation for this is to clean up expired special editions. 

The main risk with this functionality is that it ends up deleting editions which are still in use. I think this is safe, because the `editionsListCache` only ever gets set after a succesful API request to the /editions endpoint. 

@jimhunty suggested not actually deleting the directories themselves as we've noticed that cause bugs before, so this job just deletes all the issues in the root /special-edition/* directory (but deletes everything in the /download/special-edition directory.

[**Trello Card ->**](https://trello.com/c/e9O8wPMV/1558-delete-special-editions-after-they-have-expired)

## Test Plan

I've tested this in the simulator and checked that it deletes the right files. 

We could experiment on this internally by:

- switch to proof code API
- download earth edition 
- check size of app
- switch to  published code API
- restart app
- check that earth edition is no longer downloaded
